internal/charmstore/store_test.go

```diff
@@ -1166,10 +1167,32 @@ func (s *StoreSuite) TestAddCharmWithBundleSeries(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, `charm added with invalid id cs:~charmers/bundle/wordpress-2`)
 }
 
+func (s *StoreSuite) TestAddCharmWithMultipleSeries(c *gc.C) {
+	store := s.newStore(c, false)
+	defer store.Close()
+	ch := storetesting.Charms.CharmArchive(c.MkDir(), "juju-gui")
+	s.checkAddCharm(c, ch, false, newResolvedURL("~charmers/juju-gui-1", 1))
+	// Make sure it can be accessed with a number of names
+	e, err := store.FindEntity(newResolvedURL("~charmers/juju-gui-1", 1))
+	c.Assert(err, gc.IsNil)
+	c.Assert(e.URL.String(), gc.Equals, "cs:~charmers/juju-gui-1")
+	e, err = store.FindEntity(newResolvedURL("~charmers/lucid/juju-gui-1", 1))
+	c.Assert(err, gc.IsNil)
+	c.Assert(e.URL.String(), gc.Equals, "cs:~charmers/juju-gui-1")
+	e, err = store.FindEntity(newResolvedURL("~charmers/trusty/juju-gui-1", 1))
+	c.Assert(err, gc.IsNil)
+	c.Assert(e.URL.String(), gc.Equals, "cs:~charmers/juju-gui-1")
+	e, err = store.FindEntity(newResolvedURL("~charmers/wily/juju-gui-1", 1))
+	c.Assert(err, gc.IsNil)
+	c.Assert(e.URL.String(), gc.Equals, "cs:~charmers/juju-gui-1")
+	_, err = store.FindEntity(newResolvedURL("~charmers/precise/juju-gui-1", 1))
+	c.Assert(err, gc.ErrorMatches, "entity not found")
+	c.Assert(errgo.Cause(err), gc.Equals, params.ErrNotFound)
+}
```

Comment:
Missing test where old version of a charm can't be added after new version with metadata series is in the charmstore?